bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/WlanThermoNanoHandler.java

```diff
@@ -0,0 +1,245 @@
+
+    public WlanThermoNanoHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing WlanThermo Nano!");
+        config = getConfigAs(WlanThermoNanoConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        try {
+            if (config.getUsername() != null && !config.getUsername().isEmpty() && config.getPassword() != null
+                    && !config.getPassword().isEmpty()) {
+                AuthenticationStore authStore = httpClient.getAuthenticationStore();
+                authStore.addAuthentication(new DigestAuthentication(config.getUri(), Authentication.ANY_REALM,
+                        config.getUsername(), config.getPassword()));
+            }
+            httpClient.start();
+
+            scheduler.schedule(this::checkConnection, config.getPollingInterval(), TimeUnit.SECONDS);
+
+            logger.debug("Finished initializing WlanThermo Nano!");
+        } catch (Exception e) {
+            logger.error("Failed to initialize WlanThermo Nano!", e);
+        }
+    }
+
+    private void checkConnection() {
+        updateState(SYSTEM + "#" + SYSTEM_ONLINE, OnOffType.OFF);
+        try {
+            if (httpClient.GET(config.getUri()).getStatus() == 200) {
+                updateStatus(ThingStatus.ONLINE);
+                if (pollingScheduler != null) {
+                    pollingScheduler.cancel(true);
+                }
+                pollingScheduler = scheduler.scheduleWithFixedDelay(this::update, 0, config.getPollingInterval(), TimeUnit.SECONDS);
+                updateState(SYSTEM + "#" + SYSTEM_ONLINE, OnOffType.ON);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "WlanThermo not found under given address.");
+            }
+        } catch (Exception e) {
+            logger.debug("Failed to connect.", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Could not connect to WlanThermo at "+config.getIpAddress());
+            if (pollingScheduler != null) {
+                pollingScheduler.cancel(true);
+            }
+            pollingScheduler = scheduler.schedule(this::checkConnection, config.getPollingInterval(), TimeUnit.SECONDS);
+        }
+    }
+    
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            State s = data.getState(channelUID, this);
+            if (s != null)
+                updateState(channelUID, s);
+        } else {
+            if (data.setState(channelUID, command)) {
+                logger.debug("Data updated, pushing changes");
+                push();
+            } else {
+                logger.error("Could not handle command of type {} for channel {}!",command.getClass().toGenericString() , channelUID.getId());
```

Comment:
Only log to error is this is a serious bug. This looks more like a user error and should probably be a warn or info log.